pr_agent/git_providers/bitbucket_server_provider.py

```diff
@@ -38,10 +38,31 @@ def __init__(
         self.diff_files = None
         self.bitbucket_pull_request_api_url = pr_url
         self.bearer_token = get_settings().get("BITBUCKET_SERVER.BEARER_TOKEN", None)
+        #Get username and password from settings
+        self.username = get_settings().get("BITBUCKET_SERVER.USERNAME", None)
+        self.password = get_settings().get("BITBUCKET_SERVER.PASSWORD", None)
+        if not self.bitbucket_server_url:
+            raise ValueError("Invalid or missing Bitbucket Server URL parsed from PR URL.")
         self.bitbucket_server_url = self._parse_bitbucket_server(url=pr_url)
```

Comment:
/implement

First get `self.bitbucket_server_url` value and only then check it.

```
       self.bitbucket_server_url = self._parse_bitbucket_server(url=pr_url)
        if not self.bitbucket_server_url:
            raise ValueError("Invalid or missing Bitbucket Server URL parsed from PR URL.")
```